src/components/ActionsMenuLayout/ActionsMenuLayout.tsx

```diff
@@ -16,30 +20,102 @@ export class ActionsMenuLayout extends React.Component<ActionsMenuLayoutProps> {
   static Divider = ActionsMenuLayoutDivider;
 
   static displayName = 'ActionsMenuLayout';
+  state = {
+    focusedIdx: 0,
+  };
+
+  liRefs = [];
+
+  componentDidMount() {
+    const { focusedIndex } = this.props;
+
+    if (focusedIndex) {
+      this.focusItem(focusedIndex);
+    } else {
+      this.focusItem(this.state.focusedIdx);
+    }
+  }
+
+  // componentDidUpdate() {
+
+  // }
+
+  focusItem(idx: number) {
+    this.setState(
+      {
+        focusedIdx: idx,
+      },
+      () => {
+        this.liRefs[idx].liRef?.current.focus();
+      },
+    );
+  }
 
   getDataAttributes(mobile: boolean) {
     return {
       [ACTIONS_MENU_DATA_KEYS.mobile]: mobile,
     };
   }
 
+  getNextItemIdx(direction: 1 | -1): number {
+    const nextItem = this.state.focusedIdx + direction;
+
+    if (direction === 1) {
+      return nextItem === this.liRefs.length ? 0 : nextItem;
+    }
+
+    return nextItem < 0 ? this.liRefs.length - 1 : nextItem;
+  }
+
+  _onKeyDown = (e: React.KeyboardEvent<HTMLUListElement>) => {
+    if (e.key === KEYS.ArrowDown || e.key === KEYS.ArrowDownIE) {
+      e.preventDefault(); // remove scroll on space
+      this.focusItem(this.getNextItemIdx(1));
+    }
+
+    if (e.key === KEYS.ArrowUp || e.key === KEYS.ArrowUpIE) {
+      e.preventDefault(); // remove scroll on space
+      this.focusItem(this.getNextItemIdx(-1));
+    }
+  };
+
+  addRef(ref, index: number) {
+    if (ref) {
+      if (
+        this.liRefs.find(_ref => {
```

Comment:
here is a bit of "magic" bc the list of refs might be not normalized

Reply:
I understand 🤔 
This is implementation is fine, but question:
doesn't React handle this automatically?
I mean, React calls the ref only on when it mounts, and unmounts (with `null`)
So this might work out by itself, no?